src/app/code/community/Shopgate/Cloudapi/Model/Api2/Observers/WishlistsItemsRetrieve.php

```diff
@@ -43,10 +43,47 @@ public function execute(Varien_Event_Observer $observer)
         foreach ($collection as $item) {
             $data               = $item->getData();
             $data['buyRequest'] = $item->getBuyRequest()->getData();
+            $data['child_id']   = $this->getChildId($item);
             $items[]            = $data;
         }
 
         $output = $observer->getData('output');
         $output->setData('items', $items);
     }
+
+    /**
+     * Returns the product child id based on parents super attributes if exist
+     *
+     * @param Mage_Wishlist_Model_Item $item
+     *
+     * @return null | int
+     */
+    private function getChildId($item)
+    {
+        $superAttribute = $item->getBuyRequest()->getSuperAttribute();
+        if (is_array($superAttribute) && count($superAttribute) && count(array_values($superAttribute))) {
```

Comment:
any particular scenario that the other 2 statements don't cover? referring to `count(array_values($superAttribute))`